vm/src/builtins/dict.rs

```diff
@@ -782,10 +782,10 @@ macro_rules! dict_iterator {
                                 "dictionary changed size during iteration".to_owned(),
                             ));
                         }
-                        let count = zelf.position.fetch_add(1);
-                        match zelf.dict.len().checked_sub(count) {
-                            Some(mut pos) => {
-                                let (key, value) = zelf.dict.entries.next_entry(&mut pos).unwrap();
+                        let mut position = zelf.position.load();
+                        match zelf.dict.entries.next_entry_reversed(&mut position) {
+                            Some((key, value)) => {
+                                zelf.position.store(position);
```

Comment:
`fetch_add` was atomic operation, but the combination of `load` and `store ` is not. Can we express this as consecutive `fetch_add`?

Reply:
i fixed this with only using fetch_add in next_entry_reversed. there is same problem in dict iterator, i would fix it in another PR since this PR is about reverse iterator.